decidim-comments/app/queries/decidim/comments/metrics/comments_metric_manage.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Comments
+    module Metrics
+      class CommentsMetricManage < Decidim::MetricManage
+        def metric_name
+          "comments"
+        end
+
+        def save
+          return @registry if @registry
+
+          @registry = []
+          query.each do |key, results|
+            cumulative_value = results[:cumulative]
+            next if cumulative_value.zero?
+            quantity_value = results[:quantity] || 0
+            space_type, space_id, category_id, related_object_type, related_object_id = key
+            record = Decidim::Metric.find_or_initialize_by(day: @day.to_s, metric_type: @metric_name,
+                                                           participatory_space_type: space_type, participatory_space_id: space_id,
+                                                           organization: @organization, decidim_category_id: category_id,
+                                                           related_object_type: related_object_type, related_object_id: related_object_id)
+            record.assign_attributes(cumulative: cumulative_value, quantity: quantity_value)
+            @registry << record
+          end
+          @registry.each(&:save!)
+          @registry
+        end
+
+        private
+
+        def query
```

Comment:
This method is way too long, and might mean the abstraction is not OK.

Reply:
`Comments`  has a polymorphic relation, and for metrics I need to retrieve the `ParticipatorySpace`. That's why it seems weird.
I'll try to make it more readable.